src/qt/walletcontroller.cpp

```diff
@@ -108,6 +108,12 @@ WalletModel* WalletController::getOrCreateWallet(std::unique_ptr<interfaces::Wal
     wallet_model->setParent(this);
     m_wallets.push_back(wallet_model);
 
+    // WalletModel::startPollBalance needs to be called in a thread managed by
```

Comment:
Are you sure this comment is correct? Doesn't `QTimer(self)`, as the timer is intended to end up in the event loop of the GUI thread, need to be called from the GUI thread?

(the fix is correct in any case)

Reply:
> as the timer is intended to end up in the event loop of the GUI thread

Yes this is the case at the moment, but probably will be moved to other thread, like #17135 (same thread maybe doesn't matter now).

Either way `startTimer` can't be called from a RPC thread.